src/python/qepsi4/_ccsd_amplitude_parser.py

```diff
@@ -0,0 +1,303 @@
+############################################################################
+#   Copyright 2017 The OpenFermion Developers
+#   Modifications copyright 2021 Zapata Computing, Inc. for compatibility reasons.
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+############################################################################
+
+import numpy as np
+from openfermion import FermionOperator
+
+# Original function: https://github.com/quantumlib/OpenFermion-Psi4/blob/master/openfermionpsi4/_psi4_conversion_functions.py#L97
+def parse_amplitudes_from_psi4_ccsd(
```

Comment:
Can we split this function into multiple smaller ones?
It's hard to follow and probably also hard to test the way it's now.

Reply:
@mstechly Done, can you please check how it looks now?

Reply:
It looks better, but is still hard to follow. If you think you can extract something more that would be great, if not that's fine.

Reply:
But please add typing :)